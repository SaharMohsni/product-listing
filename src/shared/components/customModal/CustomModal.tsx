import React, { useState } from 'react';
import { Modal } from 'antd';

import './custom-modal.css';

export interface IOwnProps {
	title: string;
	loading: boolean;
	modalVisible: boolean;
	handleSubmit(): void;
	handleCancel(): void;
}
const CustomModal: React.FC<IOwnProps> = ({ children, title, modalVisible, handleSubmit, handleCancel, loading }) => {
	const handleSubmitAction = () => {
		handleSubmit();
	};

	const handleCancelAction = () => {
		handleCancel();
	};

	return (
		<div className="market-page-customization-section">
			<Modal
				title={title}
				visible={modalVisible}
				onOk={handleSubmitAction}
				confirmLoading={loading}
				onCancel={handleCancelAction}
			>
				{children}
			</Modal>
		</div>
	);
};

export default CustomModal;
