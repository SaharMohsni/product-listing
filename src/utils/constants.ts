export const ASC_PRICE = 0;
export const DESC_PRICE = 1;
export const NEW_ADDED = 2;
export const OLD_ADDED = 3;

export const LIMIT_PRODUCTS_BY_Page = 16;

export const sortDataMessages = {
	[ASC_PRICE]: 'Price low to high',
	[DESC_PRICE]: 'Price high to low',
	[NEW_ADDED]: 'New to old',
	[OLD_ADDED]: 'Old to new'
};

export const MIN_PRODUCT_QUANTITY_IN_BASKET = 1;
export const FAKE_DATA = Array.from({ length: 16 }).map(() => {}).map((el, i) => ({
	tags: [ '' ],
	price: i,
	name: 'string',
	description: 'string',
	slug: 'string',
	added: i,
	manufacturer: i,
	itemType: 'string'
}));
export const DELAY_TIME = 3000;
