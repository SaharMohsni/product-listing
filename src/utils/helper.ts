import { ISearchProductsPayload } from '../features/types/products.types';
import { isEmpty } from 'lodash';
export const getURLCleanPath = (pathname: string) => pathname.split('/')[1];

export const serialize = function(obj: any) {
	var str = [];
	for (var p in obj)
		if (obj.hasOwnProperty(p)) {
			str.push(encodeURIComponent(p) + '=' + encodeURIComponent(obj[p]));
		}
	return str.join('&');
};

export const handleNavigationQuery = (params: ISearchProductsPayload) => {
	let { page, sortVariable, sortType } = params;
	let query = params;
	if (!isEmpty(params) && (isEmpty(sortVariable) || isEmpty(sortType))) {
		query = {
			page: page
		};
	} else {
		query = {
			...params
		};
	}

	return `${serialize(query)}`;
};
