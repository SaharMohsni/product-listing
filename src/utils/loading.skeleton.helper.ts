export const fullRowItemSkeleton = (loading:boolean, active = true) => ({
    active,
    title: { rows: 1, width: '100%' },
    paragraph: false,
    loading,
  })
  export const labelSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: { rows: 1, width: '20%'  },
    loading,
  })
  export const tagSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: { rows: 1 },
    loading,
  })
  export const imageSkeleton = (loading:boolean, active = true ) => ({
    loading,
  active,
  paragraph: false,
  title:false
  })

  export const descriptionSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: { rows: 3, width: '100%' },
    loading,
  })
  export const shapeSquareBoxSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: false,
    loading,
  })
