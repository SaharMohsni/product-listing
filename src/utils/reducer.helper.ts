import { ISearchProductsPayload } from '../features/types/products.types';

export const handleSearchParams = (localSearchParams: ISearchProductsPayload, params: ISearchProductsPayload) => {
	let { page, limit, sortVariable, sortType } = localSearchParams;
	return {
		...localSearchParams,
		page: params.page ? params.page : page,
		limit: params.limit ? params.limit : limit,
		sortVariable: params.sortVariable ? params.sortVariable : sortVariable,
		sortType: params.sortType ? params.sortType : sortType
	};
};
