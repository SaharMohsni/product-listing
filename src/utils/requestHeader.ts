 
export const requestHeader = (options: object| null ): object => ({
  headers: { 
    // 'Access-Control-Allow-Origin': '*', 
    ...options,
  },
})

 