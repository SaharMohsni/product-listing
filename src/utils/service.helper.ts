import { isEmpty } from 'lodash';
export const addSortToUrl = (variable: string | undefined, sortType: string | undefined) => {
	if (!isEmpty(variable) && !isEmpty(sortType)) {
		return `&_sort=${variable}&_order=${sortType}`;
	} else return '';
};

export const addFilterToUrl = (variable: any | undefined) => {
	let str = [];
	if (variable) {
		for (var el in variable.data) {
			str.push(variable.key + '=' + variable.data[el]);
		}
	}
	return str.join('&');
};

export const addFilterByArrayListToUrl = (variable: any | undefined) => {
	let str = [];
	if (variable) {
		for (var el in variable.data) {
			str.push(variable.key + '_like=' + variable.data[el]);
		}
	}
	return str.join('&');
};

export const addFilterByStringToUrl = (variable: any | undefined) => {
	if (!isEmpty(variable)) {
		return `itemType=${variable}`;
	} else return '';
};
