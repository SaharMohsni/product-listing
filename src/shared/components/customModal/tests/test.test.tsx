import React from 'react';
import { render } from '../../../../utils/testing';
import CustomModal, { IOwnProps } from '../CustomModal';

describe('CustomModal', () => {
	const defaultProps: IOwnProps = {
		title: '',
		loading: false,
		modalVisible: true,
		handleSubmit() {
			return;
		},
		handleCancel() {
			return;
		}
	};
	const { container } = render(<CustomModal {...defaultProps} />);
	test('should render the basic template', async () => {
		expect(container).toMatchSnapshot();
	});
});
