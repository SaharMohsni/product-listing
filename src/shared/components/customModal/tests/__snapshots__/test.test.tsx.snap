// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal should render the basic template 1`] = `
<div>
  <div
    class="market-page-customization-section"
  />
</div>
`;
